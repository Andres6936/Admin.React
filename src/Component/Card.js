import React from 'react';

export class Card extends React.Component {
    render() {
        return <div className={"card rounded shadow-sm py-2 mb-4"}>
            {this.props.children}
        </div>
    }
}
