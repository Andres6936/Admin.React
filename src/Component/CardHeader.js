import React from 'react';

export class CardHeader extends React.Component {
    render() {
        return (
            <div className={"card-header text-primary bg-white fw-bold"}>
                {this.props.title}
                {this.props.children}
            </div>
        );
    }
}
