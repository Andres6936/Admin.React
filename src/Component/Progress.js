import React from 'react';

export class Progress extends React.Component {
    render() {
        return (
            <div className="progress mb-3">
                <div className={"progress-bar " + this.props.colorBar} role="progressbar"
                     style={{width: this.props.percentage + '%'}}
                     aria-valuenow={this.props.percentage} aria-valuemin="0" aria-valuemax="100">

                </div>
            </div>
        );
    }
}
