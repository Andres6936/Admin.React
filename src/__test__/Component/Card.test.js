import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Card} from "../../Component/Card";
import toJson from "enzyme-to-json";
import {CardBody} from "../../Component/CardBody";
import {CardHeader} from "../../Component/CardHeader";

Enzyme.configure({adapter: new Adapter()})

describe('<Card/>', function () {
    it("Snapshot for default Card component", () => {
        const wrapper = shallow(<Card/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it("Render for default Card component", () => {
        const wrapper = shallow(<Card/>);
        expect(wrapper.contains(<div className={"card rounded shadow-sm py-2 mb-4  "}/>)).toEqual(true);
    })

    it("Snapshot for Card component with CardBody child", () => {
        const wrapper = shallow((
            <Card>
                <CardBody/>
            </Card>
        ));
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it("Render the CardBody when it is passed as a child", () => {
        const wrapper = shallow((
            <Card>
                <CardBody/>
            </Card>
        ));
        expect(wrapper.contains(<CardBody/>)).toEqual(true);
    })

    it("Snapshot for Card component with CardBody and CardHeader child", () => {
        const wrapper = shallow((
            <Card>
                <CardHeader/>
                <CardBody/>
            </Card>
        ));
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it("Render the CardBody and CardHeader when its are passed as children", () => {
        const wrapper = shallow((
            <Card>
                <CardHeader/>
                <CardBody/>
            </Card>
        ));
        expect(wrapper.contains(<CardHeader/>)).toEqual(true);
        expect(wrapper.contains(<CardBody/>)).toEqual(true);
    })
});
