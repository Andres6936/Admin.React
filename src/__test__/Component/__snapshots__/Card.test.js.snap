// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> Snapshot for Card component with CardBody and CardHeader child 1`] = `
<div
  className="card rounded shadow-sm py-2 mb-4  "
>
  <CardHeader />
  <CardBody />
</div>
`;

exports[`<Card/> Snapshot for Card component with CardBody child 1`] = `
<div
  className="card rounded shadow-sm py-2 mb-4  "
>
  <CardBody />
</div>
`;

exports[`<Card/> Snapshot for default Card component 1`] = `
<div
  className="card rounded shadow-sm py-2 mb-4  "
/>
`;
